File: d2b.py
#!/usr/bin/python

def dec2bin(n):
    l = []
    while n > 0:
        if n % 2 == 0:
            l.append(0)
        else:
            l.append(1)
        n = n / 2
    #return ''.join(map(str, l))[::-1]
    return ''.join(reversed((map(str, l))))

print dec2bin(42)
print dec2bin(512)
print dec2bin(1024)
print dec2bin(1138)
print dec2bin(2016)

File: d2b.pl
#!/usr/bin/perl

sub dec2bin {
    my $n = shift;
    my @bin;
    while (int($n) > 0) {
        if ($n % 2 == 0) { 
            push @bin, 0; 
        } else {
            push @bin, 1;
        }
        $n = $n / 2; 
    }
    return reverse(@bin);
}

print dec2bin(42), "\n"; 
print dec2bin(512), "\n"; 
print dec2bin(1024), "\n"; 
print dec2bin(1138), "\n"; 
print dec2bin(2016), "\n"; 
